Tests/UnitTests/Purchasing/Purchases/PurchasesConfiguringTests.swift

```diff
@@ -92,6 +92,34 @@ class PurchasesConfiguringTests: BasePurchasesTests {
         expect(Purchases.shared.finishTransactions) == true
     }
 
+    func testUserIdIsSetWhenConfiguringWithUserID() {
+        let purchases = Purchases.configure(
+            with: .init(withAPIKey: "")
+                .with(appUserID: Self.appUserID)
+        )
+        expect(purchases.appUserID) == Self.appUserID
+    }
+
+    func testUserIdIsSetToAnonymousWhenConfiguringWithEmptyUserID() {
+        let purchases = Purchases.configure(
+            with: .init(withAPIKey: "")
+                .with(appUserID: "")
+        )
+        expect(purchases.appUserID).toNot(beEmpty())
+        expect(IdentityManager.userIsAnonymous(purchases.appUserID))
```

Comment:
Weirdly this fails because `DeviceCache` has a previously set `user ID`.
Is that expected? Shouldn’t the SDK “log out” when initializing with nil even if previously there was a user?